test/test_prototype_builtin_datasets.py

```diff
@@ -148,6 +163,28 @@ def test_save_load(self, test_home, dataset_mock, config):
             buffer.seek(0)
             assert_samples_equal(torch.load(buffer), sample)
 
+    @parametrize_dataset_mocks(DATASET_MOCKS)
+    def test_infinite_buffer_size(self, test_home, dataset_mock, config):
+        dataset_mock.prepare(test_home, config)
+        dataset = datasets.load(dataset_mock.name, **config)
+
+        for dp in extract_datapipes(dataset):
+            if isinstance(
```

Comment:
Should we just check the existence of the `buffer_size` attribute instead? Otherwise we'll have to update this list for all classes that have a buffer